### ISSUE TITLE
Allow user to specify learning rate schedule in model config

### ISSUE THREAD

**Issue by amanchoudhri** [neurostatslab/gerbilizer#27]
Would be great to include support for all the schedulers in `torch.optim.lr_scheduler` but at the very least, we should have:
 - exponential decay
 - cosine annealing (current)
 - reduce on plateau

Maybe in a config subdict 'LR_SCHEDULE_PARAMS' under 'OPTIMIZATION'?
